Content.Shared/Interaction/SharedInteractionSystem.cs

```diff
@@ -115,6 +120,29 @@ private void OnRemoveAttempt(EntityUid uid, UnremoveableComponent item, Containe
             args.Cancel();
         }
 
+        private bool HandleTryPullObject(ICommonSession? session, EntityCoordinates coords, EntityUid uid)
+        {
+            if (!ValidateClientInput(session, coords, uid, out var userEntity))
+            {
+                Logger.InfoS("system.interaction", $"TryPullObject input validation failed");
+                return true;
+            }
+
+            if (userEntity.Value == uid)
+                return false;
+
+            if (Deleted(uid))
+                return false;
+
+            if (!InRangeUnobstructed(userEntity.Value, uid, popup: true))
+                return false;
+
+            if (!TryComp(uid, out SharedPullableComponent? pull))
+                return false;
+
+            _pullSystem.TogglePull(userEntity.Value, pull);
+            return false;
```

Comment:
It seems like the function can only return false at this point, whereas the old function returned the value of `TogglePull`.

![image](https://user-images.githubusercontent.com/97707302/196523621-c3e429c3-bfca-422f-8ccb-85ef40d41b8a.png)

Are the preceding `return false;` statements still necessary?

Reply:
Good question, this actually gave me grief writing this pr!

CommandBinds have the possibility to designate that a given keypress doesnt even need to be sent to the server. Sadly this behaviour is not documented and i was only able to find it through excessive debugging which eventually landed me at a point in code where the check actually happens, in which it actually is written down, albeit a small comment in a bunch of code.
So basically: if the keypress should not even be sent to the server, the function should return true. Since this was on server till now, that line of code never made us any issues. Until now :D

As for the return making the previous returns redundant: No. If the only use of the return were to return the value, false, you would be correct. But another (even more important) use of the return is to not execute the remaining code following it. If one check fails, i dont want the other checks (or possibly even togglepull!) to still execute.

Reply:
Right! I've used this return-early method before in other areas. Makes a lot of sense, thank you.